src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java

```diff
@@ -264,10 +311,12 @@ public Response downloadZip(JsonObject content,
             Result result = controller.execute();
             if (result instanceof Failed)
             {
-               return Response.status(Status.INTERNAL_SERVER_ERROR).entity(result.getMessage()).build();
+               return Response.status(Status.INTERNAL_SERVER_ERROR).entity(getMessage(result)).build();
```

Comment:
so if you're writing a little query command that returns, say, the list of git providers:
```json
[ 'github', 'gogs' ]
```
that returns a nice blob of JSON to a front end, the above would generate a response like this:
```json
{
  status: "SUCCESS",
  message: "[ \"github\", \"gogs\", ]"
}
```

Which then means a caller then needs to extract the `message` as a String then parse it as JSON again to get the real JSON the caller wanted. We know a request succeeded as we get a 200 back. So I'm not sure what value JSON encoding a JSON string and wrapping it in another object adds really - it just makes the REST API harder to use for clients.

Note this query REST API was particularly not for wizards or multi-page requests; it was intended as just a simple REST API for front ends to grab a little blob of JSON data they can use in UIs.

How about for errors we wrap it in an envelope but just return the message as-is for success?